packages/paste-core/components/alert-dialog/__tests__/index.spec.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import {matchers} from 'jest-emotion';
+import {render, screen} from '@testing-library/react';
+// @ts-ignore typescript doesn't like js imports
+import axe from '../../../../../.jest/axe-helper';
+import {AlertDialogForNotifying, AlertDialogDestructive, AlertDialogExample} from '../stories/index.stories';
+
+expect.extend(matchers);
+
+describe('Alert Dialog', () => {
+  it('Should render an alert dialog box', () => {
+    render(<AlertDialogExample />);
+    expect(screen.getByText('Submit Application')).toBeDefined();
+  });
+
+  it('Should only have one, labeled button when no secondary label or action is given', () => {
+    render(<AlertDialogForNotifying />);
+    expect(screen.getAllByRole('button')).toHaveLength(1);
+    expect(screen.getByRole('button')).toHaveTextContent('Okay');
+  });
+
+  it('Should have two, labeled buttons when a secondary label and action is given', () => {
+    render(<AlertDialogExample />);
+    const buttons = screen.getAllByRole('button');
+    expect(buttons).toHaveLength(2);
+    expect(buttons[0]).toHaveTextContent('Cancel');
+    expect(buttons[1]).toHaveTextContent('Submit');
+  });
+
+  it('Should have a destructive button style when the destructive prop is included', () => {
+    render(<AlertDialogDestructive />);
+    expect(screen.getByText('Delete')).toHaveStyle('background-color: color-background-destructive');
+  });
+
+  it('Should have a heading the same as the heading prop', () => {
+    render(<AlertDialogExample />);
+    expect(screen.getByRole('heading')).toHaveTextContent('Submit Application');
+  });
+
+  it('Should have the correct aria attributes', () => {
+    render(<AlertDialogExample />);
+    expect(screen.getByRole('alertdialog')).toBeTruthy();
+    expect(screen.getByRole('alertdialog').getAttribute('aria-modal')).toEqual('true');
+    expect(screen.getByRole('alertdialog').getAttribute('aria-labelledby')).toEqual(
+      screen.getByRole('heading').getAttribute('id')
+    );
+    expect(screen.getByRole('alertdialog').getAttribute('aria-describedby')).toEqual(
+      screen
+        .getByText('Are you sure you want to submit this application? No information can be changed after submitting.')
+        .getAttribute('id')
+    );
+  });
+
+  it('Should have the initial focus land on the first focusable item', () => {
+    render(<AlertDialogExample />);
+    expect(document.activeElement).toEqual(screen.getAllByRole('button')[0]);
+  });
+});
+
+describe('Accessibility', () => {
+  it('Should have no accessibility violations', async () => {
+    const {container} = render(<AlertDialogExample />);
```

Comment:
This is probably sufficient, but you might want to consider adding accessibility tests for the other storybook examples that you're testing too. When I was writing my tests, I found that one of my stories was failing because the child elements were lacking unique IDs.

Reply:
ah yes yes, I'll add one for the one-button version of alert dialog (the destructive version is a styling change, so I think it would be redundant to include)